### PR TITLE
[Compute] BREAKING CHANGE: `az disk create`: Support creating disk with Gen2 and TLVM as default

### PR DESCRIPTION
**Pull request by yanzhudd**
**Related command**

`az disk create`

**Description**

Close: #23614

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 1]
```diff
@@ -420,64 +416,27 @@ def create_managed_disk(cmd, resource_group_name, disk_name, location=None,  # p
...
-                        ['TrustedLaunchSupported', 'TrustedLaunchAndConfidentialVmSupported'] \
-                        and security_type != 'TrustedLaunch':
-                    # will set default value of security_type
-                    logger.warning(log_message)
+                    hyper_v_generation = 'V2'
+                # set default value of security_type
+                if not security_type:
+                    security_type = 'TrustedLaunch'
+                if security_type != 'TrustedLaunch':
+                    logger.warning(UPGRADE_SECURITY_HINT)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:428]: > will set default value of security_type

In this case, we only need to print the warning logs and no longer modify the default values, right?

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 2]
```diff
@@ -420,64 +416,27 @@ def create_managed_disk(cmd, resource_group_name, disk_name, location=None,  # p
...
-                        and 'SecurityType' in gallery_image_info.features \
-                        and gallery_image_info.features['SecurityType'] in \
-                        ['TrustedLaunchSupported', 'TrustedLaunch', 'TrustedLaunchAndConfidentialVmSupported'] \
-                        and security_type != 'TrustedLaunch':
-                    # will set default value of security_type
-                    logger.warning(log_message)
+        if not security_type:
+            security_type = 'Standard'
+        if security_type != 'TrustedLaunch':
+            logger.warning(UPGRADE_SECURITY_HINT)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:439]: We no longer need to modify the default values based on the conditions of `hyper_v_generation` and security_type`, right?

**Comment by zhoxing-ms** [comment]
Could you re-record some related tests? Modifying the default behavior will not affect the tests in playback mode, but there may also be workflow issues caused by the breaking changes which cannot be detected by playback mode

File: src/azure-cli/azure/cli/command_modules/vm/custom.py (hunk 2 above)
Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py:439]: Hi @zhoxing-ms this change has been confirmed with Ajay:
https://github.com/Azure/azure-cli/issues/23614#issuecomment-1764369004

File: src/azure-cli/azure/cli/command_modules/vm/custom.py (hunk 1 above)
Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py:428]: Hi @zhoxing-ms  this change has been confirmed with Ajay:
https://github.com/Azure/azure-cli/issues/23614#issuecomment-1764369004

**Comment by yanzhudd** [comment]
> Could you re-record some related tests? Modifying the default behavior will not affect the tests in playback mode, but there may also be workflow issues caused by the breaking changes which cannot be detected by playback mode

addressed

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**